<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ DIANNmv</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">DIANNmv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/DIANNmv.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/DijkJel/DIANNmv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="add_contaminants.html">add_contaminants()</a></code>

        </dt>
        <dd>Adds a Potential.contaminant column to pg_matrix based on MaxQuant contaminants.txt</dd>
      </dl><dl><dt>

          <code><a href="add_iBAQ.html">add_iBAQ()</a></code>

        </dt>
        <dd>Add iBAQ intensities to report.pg_matrix file.</dd>
      </dl><dl><dt>

          <code><a href="add_maxLFQ_iBAQ.html">add_maxLFQ_iBAQ()</a></code>

        </dt>
        <dd>Adds iBAQ values based on LFQ values to SummarizedExperiment</dd>
      </dl><dl><dt>

          <code><a href="add_median_peptide_intensity.html">add_median_peptide_intensity()</a></code>

        </dt>
        <dd>Adds median peptide intensities to summerizedExperiment object</dd>
      </dl><dl><dt>

          <code><a href="add_peptide_numbers.html">add_peptide_numbers()</a></code>

        </dt>
        <dd>Add the peptide number information to pg_matrix</dd>
      </dl><dl><dt>

          <code><a href="add_standardTheme.html">add_standardTheme()</a></code>

        </dt>
        <dd>Adds standard changes to ggplot theme</dd>
      </dl><dl><dt>

          <code><a href="calculate_iBAQ.html">calculate_iBAQ()</a></code>

        </dt>
        <dd>Calculate iBAQ values from raw intensities</dd>
      </dl><dl><dt>

          <code><a href="contaminants_maxquant.html">contaminants_maxquant</a></code>

        </dt>
        <dd>Data frame with potential contaminants from MaxQuant.</dd>
      </dl><dl><dt>

          <code><a href="create_imputation_mask.html">create_imputation_mask()</a></code>

        </dt>
        <dd>Returns boolean masks specifying MAR/MNAR imputation</dd>
      </dl><dl><dt>

          <code><a href="expDesign.html">expDesign</a></code>

        </dt>
        <dd>Experimental design for example data</dd>
      </dl><dl><dt>

          <code><a href="get_DEPresults.html">get_DEPresults()</a></code>

        </dt>
        <dd>Perform differential protein expression analysis</dd>
      </dl><dl><dt>

          <code><a href="get_detection_limit.html">get_detection_limit()</a></code>

        </dt>
        <dd>Find the cutoff for MAR/MNAR classification</dd>
      </dl><dl><dt>

          <code><a href="get_genesets.html">get_genesets()</a></code>

        </dt>
        <dd>Extract a collection of gene sets from the MSigDB object</dd>
      </dl><dl><dt>

          <code><a href="get_ibaq_peptides.html">get_ibaq_peptides()</a></code>

        </dt>
        <dd>Perform in silico tryptic digest on uniprot fasta file</dd>
      </dl><dl><dt>

          <code><a href="get_intensities_prMatrix.html">get_intensities_prMatrix()</a></code>

        </dt>
        <dd>Calculate protein intensities from peptide information</dd>
      </dl><dl><dt>

          <code><a href="get_median_intensities_prMatrix.html">get_median_intensities_prMatrix()</a></code>

        </dt>
        <dd>Calculates the median peptide intensity per proteinGroup per sample.</dd>
      </dl><dl><dt>

          <code><a href="get_nPep_prMatrix.html">get_nPep_prMatrix()</a></code>

        </dt>
        <dd>Get the numer of razor/unique peptides per proteinGroup per sample</dd>
      </dl><dl><dt>

          <code><a href="get_ranked_genes.html">get_ranked_genes()</a></code>

        </dt>
        <dd>Rank proteins/genes based on fold change</dd>
      </dl><dl><dt>

          <code><a href="ibaq_peptides.html">ibaq_peptides</a></code>

        </dt>
        <dd>List with calculated theoretical tryptic peptides for mouse and human for iBAQ Peptides are calculated using 'cleaver' package with 'trypsin' set as enzyme, zero miscleavages, and peptide length 7-30.</dd>
      </dl><dl><dt>

          <code><a href="load_msigdb.html">load_msigdb()</a></code>

        </dt>
        <dd>Downloads MSigDB</dd>
      </dl><dl><dt>

          <code><a href="mixed_imputation.html">mixed_imputation()</a></code>

        </dt>
        <dd>Perform mixed imputation on summarizedExperiment object</dd>
      </dl><dl><dt>

          <code><a href="mixed_imputation_matrix.html">mixed_imputation_matrix()</a></code>

        </dt>
        <dd>Perform mixed imputation on a data matrix</dd>
      </dl><dl><dt>

          <code><a href="perform_GSEA.html">perform_GSEA()</a></code>

        </dt>
        <dd>Perform GSEA based on DEP results</dd>
      </dl><dl><dt>

          <code><a href="perform_mixed_imputation.html">perform_mixed_imputation()</a></code>

        </dt>
        <dd>Perform mixed imputation over a matrix containing data of a single condition.</dd>
      </dl><dl><dt>

          <code><a href="plotVolcano.html">plotVolcano()</a></code>

        </dt>
        <dd>create volcano plots for all comparisons present in the results file</dd>
      </dl><dl><dt>

          <code><a href="plot_DEP_barplot.html">plot_DEP_barplot()</a></code>

        </dt>
        <dd>Make bar plot showing number of differential proteins between conditions.</dd>
      </dl><dl><dt>

          <code><a href="plot_MA.html">plot_MA()</a></code>

        </dt>
        <dd>Makes MA plots for each comparison in the results data frame</dd>
      </dl><dl><dt>

          <code><a href="plot_gsea_barplot.html">plot_gsea_barplot()</a></code>

        </dt>
        <dd>Plots GSEA results as bar plot.</dd>
      </dl><dl><dt>

          <code><a href="plot_gsea_bubbleplot.html">plot_gsea_bubbleplot()</a></code>

        </dt>
        <dd>Plots GSEA data as bubble plot. Mainly useful to compare multiple gsea comparisons.</dd>
      </dl><dl><dt>

          <code><a href="plot_gsea_dotplot.html">plot_gsea_dotplot()</a></code>

        </dt>
        <dd>Plots GSEA data as a faceted dot plot showing NES, padj, and set size.</dd>
      </dl><dl><dt>

          <code><a href="plot_gsea_volcano.html">plot_gsea_volcano()</a></code>

        </dt>
        <dd>Plots GSEA data as volcano plot</dd>
      </dl><dl><dt>

          <code><a href="plot_upset.html">plot_upset()</a></code>

        </dt>
        <dd>Plots upset plot showsing overlapping significant proteins between comparisons</dd>
      </dl><dl><dt>

          <code><a href="plot_venn_diagram.html">plot_venn_diagram()</a></code>

        </dt>
        <dd>Plots a Venn diagram showing overlapping significant proteins</dd>
      </dl><dl><dt>

          <code><a href="prepare_MA_data.html">prepare_MA_data()</a></code>

        </dt>
        <dd>Prepare the data for making MA plots</dd>
      </dl><dl><dt>

          <code><a href="prepare_barplot_data.html">prepare_barplot_data()</a></code>

        </dt>
        <dd>Prepare data for plot_DEP_barplot function.</dd>
      </dl><dl><dt>

          <code><a href="prepare_gsea_data.html">prepare_gsea_data()</a></code>

        </dt>
        <dd>Prepares data for the different GSEA visualization options</dd>
      </dl><dl><dt>

          <code><a href="prepare_se.html">prepare_se()</a></code>

        </dt>
        <dd>Prepare summarizedExperiment object from diann report.pg_matrix file</dd>
      </dl><dl><dt>

          <code><a href="prepare_volcano_data.html">prepare_volcano_data()</a></code>

        </dt>
        <dd>Prepare data for volcano plots</dd>
      </dl><dl><dt>

          <code><a href="recode_sig_col.html">recode_sig_col()</a></code>

        </dt>
        <dd>Relabel the significance column in the DEP results data frame</dd>
      </dl><dl><dt>

          <code><a href="report.pg_matrix.html">report.pg_matrix</a></code>

        </dt>
        <dd>Output file of DIANN with raw proteinGroup intensities on protein level.</dd>
      </dl><dl><dt>

          <code><a href="report.pr_matrix.html">report.pr_matrix</a></code>

        </dt>
        <dd>Output file of DIANN with raw intensities on peptide level.</dd>
      </dl><dl><dt>

          <code><a href="summarize_peptide_intensities.html">summarize_peptide_intensities()</a></code>

        </dt>
        <dd>Combine different peptide variants into a single intensity</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

