<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plots the coverage of proteins. — plot_protein_coverage • DIANNmv</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plots the coverage of proteins. — plot_protein_coverage"><meta name="description" content="Plots the coverage of proteins."><meta property="og:description" content="Plots the coverage of proteins."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">DIANNmv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/DIANNmv.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/DijkJel/DIANNmv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Plots the coverage of proteins.</h1>
      <small class="dont-index">Source: <a href="https://github.com/DijkJel/DIANNmv/blob/master/R/protein_coverage.R" class="external-link"><code>R/protein_coverage.R</code></a></small>
      <div class="d-none name"><code>plot_protein_coverage.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plots the coverage of proteins.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_protein_coverage</span><span class="op">(</span></span>
<span>  <span class="va">se</span>,</span>
<span>  <span class="va">genes</span>,</span>
<span>  positions <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  zoom <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  fasta <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  organism <span class="op">=</span> <span class="st">"hs"</span>,</span>
<span>  combine_overlap <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  dodge_labels <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  scaling <span class="op">=</span> <span class="st">"centered"</span>,</span>
<span>  condition_order <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-se">se<a class="anchor" aria-label="anchor" href="#arg-se"></a></dt>
<dd><p>The SummarizedExperiment object from <a href="prepare_se.html">prepare_se</a>.</p></dd>


<dt id="arg-genes">genes<a class="anchor" aria-label="anchor" href="#arg-genes"></a></dt>
<dd><p>The gene names of the proteins that you would like to plot.</p></dd>


<dt id="arg-positions">positions<a class="anchor" aria-label="anchor" href="#arg-positions"></a></dt>
<dd><p>A numeric vector indiciting the amino acid positions that
you want to highlight.</p></dd>


<dt id="arg-zoom">zoom<a class="anchor" aria-label="anchor" href="#arg-zoom"></a></dt>
<dd><p>A numeric with two values (start and stop) indicating the specific
part of the protein that you want to plot.</p></dd>


<dt id="arg-fasta">fasta<a class="anchor" aria-label="anchor" href="#arg-fasta"></a></dt>
<dd><p>Data frame with protein information when not using a default option.
See <a href="get_ibaq_peptides.html">get_ibaq_peptides</a></p></dd>


<dt id="arg-organism">organism<a class="anchor" aria-label="anchor" href="#arg-organism"></a></dt>
<dd><p>Specifies which default protein database to use.
'hs' for human, or 'mm' for mouse.</p></dd>


<dt id="arg-combine-overlap">combine_overlap<a class="anchor" aria-label="anchor" href="#arg-combine-overlap"></a></dt>
<dd><p>Boolean specifying whether to combine miscleaved peptides.</p></dd>


<dt id="arg-dodge-labels">dodge_labels<a class="anchor" aria-label="anchor" href="#arg-dodge-labels"></a></dt>
<dd><p>Boolean specifying wheter position labels should be plotted
on different heights so that they do not overlap in the plot.</p></dd>


<dt id="arg-scaling">scaling<a class="anchor" aria-label="anchor" href="#arg-scaling"></a></dt>
<dd><p>Boolean value indicating whether the peptide intensities should
be centered over the different replicates.</p></dd>


<dt id="arg-condition-order">condition_order<a class="anchor" aria-label="anchor" href="#arg-condition-order"></a></dt>
<dd><p>Character vector that specifies the order of the conditions
on the y-axis.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A ggplot2 object showing the found peptides for a protein.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="prepare_se.html">prepare_se</a></span><span class="op">(</span><span class="va">report.pg_matrix</span>, <span class="va">expDesign</span>, <span class="va">report.pr_matrix</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plot_protein_coverage-1.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Imputing along margin 2 (samples/columns).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.3066195</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Imputing along margin 1 (features/rows).</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>36 rows with more than 50 % entries missing;</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>  mean imputation used for these rows</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cluster size 5574 broken into 3662 1912 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cluster size 3662 broken into 1462 2200 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Done cluster 1462 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cluster size 2200 broken into 1110 1090 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Done cluster 1110 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Done cluster 1090 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Done cluster 2200 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Done cluster 3662 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cluster size 1912 broken into 1379 533 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Done cluster 1379 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Done cluster 533 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Done cluster 1912 </span>
<span class="r-in"><span><span class="va">smad4</span> <span class="op">&lt;-</span> <span class="fu">plot_protein_coverage</span><span class="op">(</span><span class="va">se</span>, <span class="st">'SMAD4'</span>, positions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">150</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Duplicated aesthetics after name standardisation: <span style="color: #00BB00;">ymin</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Coordinate system already present.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Adding new coordinate system, which will replace the existing one.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Zoom in on first 200 AA of protein</span></span></span>
<span class="r-in"><span><span class="va">smad4</span> <span class="op">&lt;-</span> <span class="fu">plot_protein_coverage</span><span class="op">(</span><span class="va">se</span>, <span class="st">'SMAD4'</span>, positions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">150</span><span class="op">)</span>, zoom <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">200</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Duplicated aesthetics after name standardisation: <span style="color: #00BB00;">ymin</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Coordinate system already present.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Adding new coordinate system, which will replace the existing one.</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

