<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Prepares the data for protein coverage plots. — prepare_peptide_data • DIANNmv</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Prepares the data for protein coverage plots. — prepare_peptide_data"><meta name="description" content="Prepares the data for protein coverage plots."><meta property="og:description" content="Prepares the data for protein coverage plots."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">DIANNmv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/DIANNmv.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/DijkJel/DIANNmv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Prepares the data for protein coverage plots.</h1>
      <small class="dont-index">Source: <a href="https://github.com/DijkJel/DIANNmv/blob/master/R/protein_coverage.R" class="external-link"><code>R/protein_coverage.R</code></a></small>
      <div class="d-none name"><code>prepare_peptide_data.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Prepares the data for protein coverage plots.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prepare_peptide_data</span><span class="op">(</span></span>
<span>  <span class="va">pr_matrix</span>,</span>
<span>  <span class="va">genes</span>,</span>
<span>  positions <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  zoom <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  fasta <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  organism <span class="op">=</span> <span class="st">"hs"</span>,</span>
<span>  combine_overlap <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  dodge_labels <span class="op">=</span> <span class="cn">T</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-pr-matrix">pr_matrix<a class="anchor" aria-label="anchor" href="#arg-pr-matrix"></a></dt>
<dd><p>The report.pr_matrix file.</p></dd>


<dt id="arg-genes">genes<a class="anchor" aria-label="anchor" href="#arg-genes"></a></dt>
<dd><p>The gene names of the proteins that you would like to plot.</p></dd>


<dt id="arg-positions">positions<a class="anchor" aria-label="anchor" href="#arg-positions"></a></dt>
<dd><p>A numeric vector indiciting the amino acid positions that
you want to highlight.</p></dd>


<dt id="arg-zoom">zoom<a class="anchor" aria-label="anchor" href="#arg-zoom"></a></dt>
<dd><p>A numeric with two values (start and stop) indicating the specific
part of the protein that you want to plot.</p></dd>


<dt id="arg-fasta">fasta<a class="anchor" aria-label="anchor" href="#arg-fasta"></a></dt>
<dd><p>Data frame with protein information when not using a default option.
See <a href="get_ibaq_peptides.html">get_ibaq_peptides</a></p></dd>


<dt id="arg-organism">organism<a class="anchor" aria-label="anchor" href="#arg-organism"></a></dt>
<dd><p>Specifies which default protein database to use.
'hs' for human, or 'mm' for mouse.</p></dd>


<dt id="arg-combine-overlap">combine_overlap<a class="anchor" aria-label="anchor" href="#arg-combine-overlap"></a></dt>
<dd><p>Boolean specifying whether to combine miscleaved peptides.</p></dd>


<dt id="arg-dodge-labels">dodge_labels<a class="anchor" aria-label="anchor" href="#arg-dodge-labels"></a></dt>
<dd><p>Boolean specifying wheter position labels should be plotted
on different heights so that they do not overlap in the plot.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list containing a data frame with information about the tryptic
peptides, and a data frame with the AA positions that are highlighted.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">coverage_data</span> <span class="op">&lt;-</span> <span class="fu">prepare_peptide_data</span><span class="op">(</span><span class="va">report.pr_matrix</span>,  <span class="st">'SMAD4'</span>, positions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">120</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jelmer Dijkstra.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

