<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>DIANNmv • DIANNmv</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="DIANNmv">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">DIANNmv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/DIANNmv.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/DijkJel/DIANNmv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>DIANNmv</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/DijkJel/DIANNmv/blob/master/vignettes/DIANNmv.Rmd" class="external-link"><code>vignettes/DIANNmv.Rmd</code></a></small>
      <div class="d-none name"><code>DIANNmv.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This package can be used to process and analyze mass spec data
generated by DIANN. It mainly uses the DEP package under the hood, so
you can use all DEP functions once you have created the
summarizedExperiment object (check out the DEP vignette). Also included
are some visualization function that I prefer over the default DEP
options.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install and load the package from github with the following code (you
can ignore the warnings):</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">'BiocManager'</a></span>, quietly <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">'devtools'</a></span>, quietly <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'devtools'</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">install_github</span><span class="op">(</span><span class="st">'DijkJel/DIANNmv'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/DijkJel/DIANNmv" class="external-link">DIANNmv</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'SummarizedExperiment::start' by</span></span>
<span><span class="co">#&gt; 'stats::start' when loading 'DIANNmv'</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'SummarizedExperiment::end' by 'stats::end'</span></span>
<span><span class="co">#&gt; when loading 'DIANNmv'</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'MsCoreUtils::smooth' by 'stats::smooth'</span></span>
<span><span class="co">#&gt; when loading 'DIANNmv'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; Loading required package: generics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'generics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,</span></span>
<span><span class="co">#&gt;     setequal, union</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,</span></span>
<span><span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">#&gt;     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,</span></span>
<span><span class="co">#&gt;     unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="required-input-data-example-data">Required input data &amp; example data<a class="anchor" aria-label="anchor" href="#required-input-data-example-data"></a>
</h2>
<p>As input files you will need the protein (report.pg_matrix) and
peptide (report.pr_matrix) DIANN output files. In addition, you need an
experimental design table. This package comes with these files from a
DNA pull down experiment for demonstration goals.You can inspect the
data with ‘expDesign’, ‘report.pg_matrix’, and’report.pr_matrix’.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">report.pg_matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  Protein.Group                      Protein.Names</span></span>
<span><span class="co">#&gt; 1 A0A075B6H7;A0A0C4DH55;P01624 KV315_HUMAN;KV37_HUMAN;KVD07_HUMAN</span></span>
<span><span class="co">#&gt; 2            A0A0A0MRZ8;P04433            KV311_HUMAN;KVD11_HUMAN</span></span>
<span><span class="co">#&gt; 3            A0A0B4J2D5;P0DPI2            GAL3A_HUMAN;GAL3B_HUMAN</span></span>
<span><span class="co">#&gt; 4                       A0A183                        LCE6A_HUMAN</span></span>
<span><span class="co">#&gt; 5                   A0A1B0GTR4                        SPRR5_HUMAN</span></span>
<span><span class="co">#&gt; 6                   A0A1B0GTU1                        ZC11B_HUMAN</span></span>
<span><span class="co">#&gt;                       Genes</span></span>
<span><span class="co">#&gt; 1 IGKV3-15;IGKV3-7;IGKV3D-7</span></span>
<span><span class="co">#&gt; 2        IGKV3-11;IGKV3D-11</span></span>
<span><span class="co">#&gt; 3              GATD3;GATD3B</span></span>
<span><span class="co">#&gt; 4                     LCE6A</span></span>
<span><span class="co">#&gt; 5                     SPRR5</span></span>
<span><span class="co">#&gt; 6                   ZC3H11B</span></span>
<span><span class="co">#&gt;                                                                      First.Protein.Description</span></span>
<span><span class="co">#&gt; 1                                    Probable non-functional immunoglobulin kappa variable 3-7</span></span>
<span><span class="co">#&gt; 2                                                          Immunoglobulin kappa variable 3D-11</span></span>
<span><span class="co">#&gt; 3 Putative glutamine amidotransferase-like class 1 domain-containing protein 3B, mitochondrial</span></span>
<span><span class="co">#&gt; 4                                                           Late cornified envelope protein 6A</span></span>
<span><span class="co">#&gt; 5                                                        Putative small proline-rich protein 5</span></span>
<span><span class="co">#&gt; 6                                               Zinc finger CCCH domain-containing protein 11B</span></span>
<span><span class="co">#&gt;   neg_ctrl_1 neg_ctrl_2 neg_ctrl_3 motif1_1 motif1_2 motif1_3 motif2_1 motif2_2</span></span>
<span><span class="co">#&gt; 1    35045.4         NA         NA  60838.8 56361.50  45830.7       NA  41375.0</span></span>
<span><span class="co">#&gt; 2    13531.4         NA         NA  26054.8       NA  50914.5  20547.4       NA</span></span>
<span><span class="co">#&gt; 3    50098.9   36549.70    43047.0  29253.3 25503.00  47893.5  36321.2  25036.7</span></span>
<span><span class="co">#&gt; 4    92060.5   92580.80         NA  49035.6 67543.30  33898.6       NA  33731.9</span></span>
<span><span class="co">#&gt; 5    57016.4   58424.50         NA 215301.0 77330.30       NA       NA  69601.4</span></span>
<span><span class="co">#&gt; 6    10773.4    9745.44    17668.3  16804.8  9534.48  17388.9  16438.0  25568.2</span></span>
<span><span class="co">#&gt;   motif2_3</span></span>
<span><span class="co">#&gt; 1 155669.0</span></span>
<span><span class="co">#&gt; 2  66331.5</span></span>
<span><span class="co">#&gt; 3  28612.7</span></span>
<span><span class="co">#&gt; 4       NA</span></span>
<span><span class="co">#&gt; 5  47255.0</span></span>
<span><span class="co">#&gt; 6       NA</span></span>
<span><span class="co">#</span></span>
<span><span class="co">#</span></span>
<span><span class="co">#</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">report.pr_matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Protein.Group   Protein.Ids Protein.Names  Genes</span></span>
<span><span class="co">#&gt; 1        Q86U42        Q86U42   PABP2_HUMAN PABPN1</span></span>
<span><span class="co">#&gt; 2        Q8NFD5 Q8NFD5;Q9Y651   ARI1B_HUMAN ARID1B</span></span>
<span><span class="co">#&gt; 3        Q96JP5        Q96JP5   ZFP91_HUMAN  ZFP91</span></span>
<span><span class="co">#&gt; 4        Q16585        Q16585    SGCB_HUMAN   SGCB</span></span>
<span><span class="co">#&gt; 5        P36578        P36578     RL4_HUMAN   RPL4</span></span>
<span><span class="co">#&gt; 6        P36578        P36578     RL4_HUMAN   RPL4</span></span>
<span><span class="co">#&gt;                          First.Protein.Description Proteotypic</span></span>
<span><span class="co">#&gt; 1                  Polyadenylate-binding protein 2           1</span></span>
<span><span class="co">#&gt; 2 AT-rich interactive domain-containing protein 1B           0</span></span>
<span><span class="co">#&gt; 3                E3 ubiquitin-protein ligase ZFP91           1</span></span>
<span><span class="co">#&gt; 4                                 Beta-sarcoglycan           1</span></span>
<span><span class="co">#&gt; 5                         60S ribosomal protein L4           1</span></span>
<span><span class="co">#&gt; 6                         60S ribosomal protein L4           1</span></span>
<span><span class="co">#&gt;     Stripped.Sequence   Modified.Sequence Precursor.Charge         Precursor.Id</span></span>
<span><span class="co">#&gt; 1    AAAAAAAAAAGAAGGR    AAAAAAAAAAGAAGGR                2    AAAAAAAAAAGAAGGR2</span></span>
<span><span class="co">#&gt; 2         AAAAAAAAAAR         AAAAAAAAAAR                2         AAAAAAAAAAR2</span></span>
<span><span class="co">#&gt; 3        AAAAAAAAAVSR        AAAAAAAAAVSR                2        AAAAAAAAAVSR2</span></span>
<span><span class="co">#&gt; 4 AAAAAAAAEQQSSNGPVKK AAAAAAAAEQQSSNGPVKK                3 AAAAAAAAEQQSSNGPVKK3</span></span>
<span><span class="co">#&gt; 5         AAAAAAALQAK         AAAAAAALQAK                1         AAAAAAALQAK1</span></span>
<span><span class="co">#&gt; 6         AAAAAAALQAK         AAAAAAALQAK                2         AAAAAAALQAK2</span></span>
<span><span class="co">#&gt;   neg_ctrl_1 neg_ctrl_2 neg_ctrl_3    motif1_1   motif1_2   motif1_3</span></span>
<span><span class="co">#&gt; 1    14201.4    17402.1    22342.6     9617.31    12355.2    10824.8</span></span>
<span><span class="co">#&gt; 2 56500900.0 59560000.0 64613700.0 51690200.00 23066300.0 40866800.0</span></span>
<span><span class="co">#&gt; 3 12186100.0 12535000.0 15420900.0 28513800.00 32014600.0 43538700.0</span></span>
<span><span class="co">#&gt; 4         NA         NA         NA     8669.59         NA         NA</span></span>
<span><span class="co">#&gt; 5  3729720.0  4311790.0  2780390.0  3405560.00  4964600.0  4373450.0</span></span>
<span><span class="co">#&gt; 6 71649900.0 31629500.0 43972100.0 67226000.00 48326400.0 47199700.0</span></span>
<span><span class="co">#&gt;      motif2_1 motif2_2    motif2_3</span></span>
<span><span class="co">#&gt; 1    11469.50    11064     9728.03</span></span>
<span><span class="co">#&gt; 2 30508900.00 30926500 39211400.00</span></span>
<span><span class="co">#&gt; 3 41882400.00 53996100 30695300.00</span></span>
<span><span class="co">#&gt; 4     8965.43       NA          NA</span></span>
<span><span class="co">#&gt; 5  6242340.00  3104850  6942140.00</span></span>
<span><span class="co">#&gt; 6 32204200.00 40380400 39495700.00</span></span>
<span><span class="co">#</span></span>
<span><span class="co">#</span></span>
<span><span class="co">#</span></span>
<span><span class="va">expDesign</span></span>
<span><span class="co">#&gt;        label condition replicate</span></span>
<span><span class="co">#&gt; 1 neg_ctrl_1  neg_ctrl         1</span></span>
<span><span class="co">#&gt; 2 neg_ctrl_2  neg_ctrl         2</span></span>
<span><span class="co">#&gt; 3 neg_ctrl_3  neg_ctrl         3</span></span>
<span><span class="co">#&gt; 4   motif1_1    motif1         1</span></span>
<span><span class="co">#&gt; 5   motif1_2    motif1         2</span></span>
<span><span class="co">#&gt; 6   motif1_3    motif1         3</span></span>
<span><span class="co">#&gt; 7   motif2_1    motif2         1</span></span>
<span><span class="co">#&gt; 8   motif2_2    motif2         2</span></span>
<span><span class="co">#&gt; 9   motif2_3    motif2         3</span></span></code></pre></div>
<p>As you can see in expDesign, there are three conditions with 3
replicates each: Two variations of a TF binding motif and one negative
control that can be used for both motifs. The names in the ‘label’
column correspond to the intensity columns in report.pg_matrix and
should be in the same order and identically named.</p>
<div class="section level3">
<h3 id="prepare-the-summarized-experiment-object">Prepare the Summarized Experiment object<a class="anchor" aria-label="anchor" href="#prepare-the-summarized-experiment-object"></a>
</h3>
<p>The report.pg_matrix, report.pr_matrix and expDesign are combined in
a summarizedExperiment (se) object that is used for downstream analysis
and plotting.</p>
<p>To create an se object, you can run the prepare_se() function with
the report.pg_matrix file and associated experimental design. You can
specify if and what type of imputation is done. The default is ‘knn’,
which is better for DIA data I think. For DDA data, ‘MinProb’ would be
the good. When ‘none’ is entered for the impute parameter, no imputation
is done. Additionally, you can filter on missing values (missing_thr),
and potential contaminants are removed by default. (Source:
contaminants.txt file from maxquant).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_se.html">prepare_se</a></span><span class="op">(</span><span class="va">report.pg_matrix</span>, <span class="va">expDesign</span><span class="op">)</span> <span class="co"># without peptide information</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Warning in knnimp(x, k, maxmiss = rowmax, maxp = maxp): 36 rows with more than 50 % entries missing;</span></span>
<span><span class="co">#&gt;  mean imputation used for these rows</span></span>
<span><span class="co">#&gt; Cluster size 5548 broken into 2022 3526 </span></span>
<span><span class="co">#&gt; Cluster size 2022 broken into 673 1349 </span></span>
<span><span class="co">#&gt; Done cluster 673 </span></span>
<span><span class="co">#&gt; Done cluster 1349 </span></span>
<span><span class="co">#&gt; Done cluster 2022 </span></span>
<span><span class="co">#&gt; Cluster size 3526 broken into 1495 2031 </span></span>
<span><span class="co">#&gt; Done cluster 1495 </span></span>
<span><span class="co">#&gt; Cluster size 2031 broken into 1030 1001 </span></span>
<span><span class="co">#&gt; Done cluster 1030 </span></span>
<span><span class="co">#&gt; Done cluster 1001 </span></span>
<span><span class="co">#&gt; Done cluster 2031 </span></span>
<span><span class="co">#&gt; Done cluster 3526</span></span>
<span><span class="va">se</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 5584 9 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): ''</span></span>
<span><span class="co">#&gt; rownames(5584): A2M A2ML1 ... ZYX ZZZ3</span></span>
<span><span class="co">#&gt; rowData names(9): Protein.Group Protein.Names ... imputed num_NAs</span></span>
<span><span class="co">#&gt; colnames(9): neg_ctrl_1 neg_ctrl_2 ... motif2_2 motif2_3</span></span>
<span><span class="co">#&gt; colData names(4): label ID condition replicate</span></span></code></pre>
<p>If you provide the report.pr_matrix alongside the report.pg_matrix,
peptide information is automatically added and you can filter on a
minimal number of razor/unique peptides (min_pep, default = 0). Make
sure the sample column names are identical and match the label column in
the experimental design.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add peptide information and remove all proteinGroups with &lt;2 total </span></span>
<span><span class="co"># razor/unique peptides</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_se.html">prepare_se</a></span><span class="op">(</span><span class="va">report.pg_matrix</span>, <span class="va">expDesign</span>, <span class="va">report.pr_matrix</span>, min_pep <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Warning in knnimp(x, k, maxmiss = rowmax, maxp = maxp): 19 rows with more than 50 % entries missing;</span></span>
<span><span class="co">#&gt;  mean imputation used for these rows</span></span>
<span><span class="co">#&gt; Cluster size 5237 broken into 2152 3085 </span></span>
<span><span class="co">#&gt; Cluster size 2152 broken into 1645 507 </span></span>
<span><span class="co">#&gt; Cluster size 1645 broken into 656 989 </span></span>
<span><span class="co">#&gt; Done cluster 656 </span></span>
<span><span class="co">#&gt; Done cluster 989 </span></span>
<span><span class="co">#&gt; Done cluster 1645 </span></span>
<span><span class="co">#&gt; Done cluster 507 </span></span>
<span><span class="co">#&gt; Done cluster 2152 </span></span>
<span><span class="co">#&gt; Cluster size 3085 broken into 1977 1108 </span></span>
<span><span class="co">#&gt; Cluster size 1977 broken into 976 1001 </span></span>
<span><span class="co">#&gt; Done cluster 976 </span></span>
<span><span class="co">#&gt; Done cluster 1001 </span></span>
<span><span class="co">#&gt; Done cluster 1977 </span></span>
<span><span class="co">#&gt; Done cluster 1108 </span></span>
<span><span class="co">#&gt; Done cluster 3085</span></span>
<span><span class="va">se</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 5256 9 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): '' peptide_info</span></span>
<span><span class="co">#&gt; rownames(5256): A2M A2ML1 ... ZYX ZZZ3</span></span>
<span><span class="co">#&gt; rowData names(11): Protein.Group Protein.Names ... imputed num_NAs</span></span>
<span><span class="co">#&gt; colnames(9): neg_ctrl_1 neg_ctrl_2 ... motif2_2 motif2_3</span></span>
<span><span class="co">#&gt; colData names(4): label ID condition replicate</span></span></code></pre>
<p>The summarizedExperiment object stores a lot of information. As you
can see from the output above, it consists of 5584 proteinGroups (rows)
and 9 samples (columns). Furthermore, the experimental design is stored
as ‘colData’, and extra information is stored as ‘rowData’. The log2
transformed intensities form the main assay. Furthermore, if the
report.pr_matrix file was provided, a second assay is added. For a
detailed description of the structure of summarizedExperiments, check
its documentation. In short, to access different parts of data:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">intensities</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span> <span class="co"># log2 protein intensities</span></span>
<span><span class="va">peptides</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">'peptide_info'</span><span class="op">)</span> <span class="co"># peptide numbers</span></span>
<span></span>
<span><span class="va">rd</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rd</span><span class="op">)</span> <span class="co"># Information for each proteinGroup in the se.</span></span>
<span><span class="co">#&gt;  [1] "Protein.Group"             "Protein.Names"            </span></span>
<span><span class="co">#&gt;  [3] "Genes"                     "First.Protein.Description"</span></span>
<span><span class="co">#&gt;  [5] "n_total"                   "Potential.contaminant"    </span></span>
<span><span class="co">#&gt;  [7] "name"                      "ID"                       </span></span>
<span><span class="co">#&gt;  [9] "npep_total"                "imputed"                  </span></span>
<span><span class="co">#&gt; [11] "num_NAs"</span></span>
<span></span>
<span><span class="va">cd</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="va">cd</span> <span class="co"># The experimental design</span></span>
<span><span class="co">#&gt;                 label         ID condition replicate</span></span>
<span><span class="co">#&gt; neg_ctrl_1 neg_ctrl_1 neg_ctrl_1  neg_ctrl         1</span></span>
<span><span class="co">#&gt; neg_ctrl_2 neg_ctrl_2 neg_ctrl_2  neg_ctrl         2</span></span>
<span><span class="co">#&gt; neg_ctrl_3 neg_ctrl_3 neg_ctrl_3  neg_ctrl         3</span></span>
<span><span class="co">#&gt; motif1_1     motif1_1   motif1_1    motif1         1</span></span>
<span><span class="co">#&gt; motif1_2     motif1_2   motif1_2    motif1         2</span></span>
<span><span class="co">#&gt; motif1_3     motif1_3   motif1_3    motif1         3</span></span>
<span><span class="co">#&gt; motif2_1     motif2_1   motif2_1    motif2         1</span></span>
<span><span class="co">#&gt; motif2_2     motif2_2   motif2_2    motif2         2</span></span>
<span><span class="co">#&gt; motif2_3     motif2_3   motif2_3    motif2         3</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="add-peptide-and-ibaq-information">Add peptide and iBAQ information<a class="anchor" aria-label="anchor" href="#add-peptide-and-ibaq-information"></a>
</h3>
<p>The report.pr_matrix and report.pg_matrix files can be used to add
peptide number information, iBAQ values, and median peptide intensities,
which are an alternative to iBAQ.</p>
<div class="section level4">
<h4 id="add-number-of-razorunique-peptides">Add number of razor/unique peptides<a class="anchor" aria-label="anchor" href="#add-number-of-razorunique-peptides"></a>
</h4>
<p>This information is automatically added to the se object when running
prepare_se() and providing a report.pr_matrix file (see above). You can
also manually add this to the report.pg_matrix() file with the following
code.</p>
<p>To get the number of razor/unique peptides per proteinGroup per
sample and the total number of razor/unique peptides over all samples,
and add these to the report.pg_matrix file:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peptides</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_nPep_prMatrix.html">get_nPep_prMatrix</a></span><span class="op">(</span><span class="va">report.pr_matrix</span><span class="op">)</span></span>
<span><span class="va">pg_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_peptide_numbers.html">add_peptide_numbers</a></span><span class="op">(</span><span class="va">report.pg_matrix</span>, <span class="va">peptides</span><span class="op">)</span></span></code></pre></div>
<p>This adds new columns with the suffix ‘npep’ to the report.pg_matrix
with the number of identified peptides per sample, and ‘n_total’ with
the total number of identified razor/unique peptides per
proteinGroup.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pg_matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Protein.Group"             "Protein.Names"            </span></span>
<span><span class="co">#&gt;  [3] "Genes"                     "First.Protein.Description"</span></span>
<span><span class="co">#&gt;  [5] "neg_ctrl_1"                "neg_ctrl_2"               </span></span>
<span><span class="co">#&gt;  [7] "neg_ctrl_3"                "motif1_1"                 </span></span>
<span><span class="co">#&gt;  [9] "motif1_2"                  "motif1_3"                 </span></span>
<span><span class="co">#&gt; [11] "motif2_1"                  "motif2_2"                 </span></span>
<span><span class="co">#&gt; [13] "motif2_3"                  "neg_ctrl_1_npep"          </span></span>
<span><span class="co">#&gt; [15] "neg_ctrl_2_npep"           "neg_ctrl_3_npep"          </span></span>
<span><span class="co">#&gt; [17] "motif1_1_npep"             "motif1_2_npep"            </span></span>
<span><span class="co">#&gt; [19] "motif1_3_npep"             "motif2_1_npep"            </span></span>
<span><span class="co">#&gt; [21] "motif2_2_npep"             "motif2_3_npep"            </span></span>
<span><span class="co">#&gt; [23] "n_total"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="add-ibaq-data">Add iBAQ data<a class="anchor" aria-label="anchor" href="#add-ibaq-data"></a>
</h4>
<p>iBAQ values can be added to the pg_matrix file as well. For that, we
need the number of theoretically observable peptides, and non-normalized
intensities.</p>
<p>The number of iBAQ peptides for reviewed human and mouse uniprot
entries are included in the package and can be accessed with:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ibaq_peptides</span> <span class="op">&lt;-</span> <span class="fu">DIANNmv</span><span class="fu">::</span><span class="va"><a href="../reference/ibaq_peptides.html">ibaq_peptides</a></span></span>
<span><span class="va">hs</span> <span class="op">&lt;-</span> <span class="va">ibaq_peptides</span><span class="op">$</span><span class="va">hs</span> <span class="co">#Human entries</span></span>
<span><span class="va">mm</span> <span class="op">&lt;-</span> <span class="va">ibaq_peptides</span><span class="op">$</span><span class="va">mm</span> <span class="co"># Mouse entries</span></span></code></pre></div>
<p>If you do not want to use these, you can create an similar object
yourself from a fasta file:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">no_ibaq_peptides</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ibaq_peptides.html">get_ibaq_peptides</a></span><span class="op">(</span><span class="st">'path/to/fasta.fasta'</span><span class="op">)</span></span></code></pre></div>
<p>Non-normalized intensities can be calculated as sum of individual
peptide intensites from the report.pr_matrix file:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">intensities</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_intensities_prMatrix.html">get_intensities_prMatrix</a></span><span class="op">(</span><span class="va">report.pr_matrix</span><span class="op">)</span></span></code></pre></div>
<p>This is all combined in the the add_iBAQ() function, which addes
columns with iBAQ values and a column with the number of iBAQ peptides
to the report.pg_matrix. If you use the included iBAQ peptides, you only
have to specify the organism. If you use a custom file, you have to
specify it with the ‘ibaq_stats’ parameter.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_iBAQ.html">add_iBAQ</a></span><span class="op">(</span><span class="va">report.pg_matrix</span>, <span class="va">report.pr_matrix</span>, organism <span class="op">=</span> <span class="st">'hs'</span><span class="op">)</span> <span class="co"># Standard</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pg</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Protein.Group"             "Protein.Names"            </span></span>
<span><span class="co">#&gt;  [3] "Genes"                     "First.Protein.Description"</span></span>
<span><span class="co">#&gt;  [5] "neg_ctrl_1"                "neg_ctrl_2"               </span></span>
<span><span class="co">#&gt;  [7] "neg_ctrl_3"                "motif1_1"                 </span></span>
<span><span class="co">#&gt;  [9] "motif1_2"                  "motif1_3"                 </span></span>
<span><span class="co">#&gt; [11] "motif2_1"                  "motif2_2"                 </span></span>
<span><span class="co">#&gt; [13] "motif2_3"                  "neg_ctrl_1_iBAQ"          </span></span>
<span><span class="co">#&gt; [15] "neg_ctrl_2_iBAQ"           "neg_ctrl_3_iBAQ"          </span></span>
<span><span class="co">#&gt; [17] "motif1_1_iBAQ"             "motif1_2_iBAQ"            </span></span>
<span><span class="co">#&gt; [19] "motif1_3_iBAQ"             "motif2_1_iBAQ"            </span></span>
<span><span class="co">#&gt; [21] "motif2_2_iBAQ"             "motif2_3_iBAQ"            </span></span>
<span><span class="co">#&gt; [23] "ibaq_peptides"</span></span></code></pre></div>
<p><br><br> The iBAQ values are automatically added to the
summarizedExperiment object when running the prepare_se() function. In
addition, the number of iBAQ peptides is added to the rowData.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_iBAQ.html">add_iBAQ</a></span><span class="op">(</span><span class="va">report.pg_matrix</span>, <span class="va">report.pr_matrix</span>, organism <span class="op">=</span> <span class="st">'hs'</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_se.html">prepare_se</a></span><span class="op">(</span><span class="va">pg</span>, <span class="va">expDesign</span>, <span class="va">report.pr_matrix</span><span class="op">)</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Warning in knnimp(x, k, maxmiss = rowmax, maxp = maxp): 33 rows with more than 50 % entries missing;</span></span>
<span><span class="co">#&gt;  mean imputation used for these rows</span></span>
<span><span class="co">#&gt; Cluster size 5475 broken into 3183 2292 </span></span>
<span><span class="co">#&gt; Cluster size 3183 broken into 2053 1130 </span></span>
<span><span class="co">#&gt; Cluster size 2053 broken into 1034 1019 </span></span>
<span><span class="co">#&gt; Done cluster 1034 </span></span>
<span><span class="co">#&gt; Done cluster 1019 </span></span>
<span><span class="co">#&gt; Done cluster 2053 </span></span>
<span><span class="co">#&gt; Done cluster 1130 </span></span>
<span><span class="co">#&gt; Done cluster 3183 </span></span>
<span><span class="co">#&gt; Cluster size 2292 broken into 1624 668 </span></span>
<span><span class="co">#&gt; Cluster size 1624 broken into 716 908 </span></span>
<span><span class="co">#&gt; Done cluster 716 </span></span>
<span><span class="co">#&gt; Done cluster 908 </span></span>
<span><span class="co">#&gt; Done cluster 1624 </span></span>
<span><span class="co">#&gt; Done cluster 668 </span></span>
<span><span class="co">#&gt; Done cluster 2292</span></span>
<span><span class="va">iBAQ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">'iBAQ'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">iBAQ</span><span class="op">)</span></span>
<span><span class="co">#&gt;       neg_ctrl_1  neg_ctrl_2 neg_ctrl_3   motif1_1  motif1_2  motif1_3</span></span>
<span><span class="co">#&gt; A2M    44908.143  43355.8406   64753.94  41528.739 123416.60  21931.53</span></span>
<span><span class="co">#&gt; A2ML1 112836.227 154355.9719   88353.84 136172.283 135247.20 156184.82</span></span>
<span><span class="co">#&gt; AAAS  595275.548 778864.7519  733462.15 248484.556 725698.24 489673.48</span></span>
<span><span class="co">#&gt; AAK1   24854.583    126.1392   26950.56  23150.526  19485.50  22769.03</span></span>
<span><span class="co">#&gt; AAR2   29334.317 183248.0889   39214.12   3291.122  35568.44  33472.99</span></span>
<span><span class="co">#&gt; AARS1   3359.781  10436.0490       0.00      0.000   2132.97   2367.20</span></span>
<span><span class="co">#&gt;         motif2_1  motif2_2   motif2_3</span></span>
<span><span class="co">#&gt; A2M     4432.100 108122.55  85534.064</span></span>
<span><span class="co">#&gt; A2ML1  47171.734  77564.26 106892.719</span></span>
<span><span class="co">#&gt; AAAS  278873.133 263530.63 505940.196</span></span>
<span><span class="co">#&gt; AAK1       0.000      0.00  21978.459</span></span>
<span><span class="co">#&gt; AAR2  201094.172 108991.24 177086.150</span></span>
<span><span class="co">#&gt; AARS1   1908.576   3763.05   2011.761</span></span>
<span></span>
<span><span class="va">ibaq_pep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">$</span><span class="va">ibaq_peptides</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ibaq_pep</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 69 64 27 36 18 50</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="add-median-peptide-intensities">Add median peptide intensities<a class="anchor" aria-label="anchor" href="#add-median-peptide-intensities"></a>
</h4>
<p>Similarly, median peptide intensities (mpi) can be added. To get the
mpi as separate object:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mpi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_median_intensities_prMatrix.html">get_median_intensities_prMatrix</a></span><span class="op">(</span><span class="va">report.pr_matrix</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mpi</span><span class="op">)</span></span>
<span><span class="co">#&gt;                        protein neg_ctrl_1 neg_ctrl_2 neg_ctrl_3  motif1_1</span></span>
<span><span class="co">#&gt; 1 A0A075B6H7;A0A0C4DH55;P01624    35045.4         NA         NA  60838.90</span></span>
<span><span class="co">#&gt; 2            A0A0A0MRZ8;P04433    13531.4    5577.47         NA  26054.80</span></span>
<span><span class="co">#&gt; 3            A0A0B4J2D5;P0DPI2    31083.5   23493.25    17767.3  18834.27</span></span>
<span><span class="co">#&gt; 4                       A0A183    74635.8   96073.55         NA  64140.95</span></span>
<span><span class="co">#&gt; 5                   A0A1B0GTR4    56976.9   58384.00         NA 148435.10</span></span>
<span><span class="co">#&gt; 6                   A0A1B0GTU1   255493.0    9745.44    17668.3  16804.80</span></span>
<span><span class="co">#&gt;    motif1_2 motif1_3 motif2_1 motif2_2  motif2_3</span></span>
<span><span class="co">#&gt; 1  56361.40  45830.7       NA  41375.0 155669.00</span></span>
<span><span class="co">#&gt; 2  25558.50  50914.5  20547.4       NA  66331.50</span></span>
<span><span class="co">#&gt; 3 249944.00  21172.3  25210.3 192574.0 267984.40</span></span>
<span><span class="co">#&gt; 4  78935.65  82479.8       NA  78484.7        NA</span></span>
<span><span class="co">#&gt; 5  47525.95       NA  65137.5  71409.6  23997.25</span></span>
<span><span class="co">#&gt; 6   9534.49  17388.9  16438.0  25568.2   8499.39</span></span></code></pre></div>
<p>To add mpi to the se object directly:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span> <span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_median_peptide_intensity.html">add_median_peptide_intensity</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">report.pr_matrix</span><span class="op">)</span></span>
<span> <span class="va">se</span> <span class="co"># an extra assay 'median_peptide_intensities' is added</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 5508 9 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(5): '' iBAQ maxLFQ_iBAQ peptide_info median_peptide_intensities</span></span>
<span><span class="co">#&gt; rownames(5508): A2M A2ML1 ... ZYX ZZZ3</span></span>
<span><span class="co">#&gt; rowData names(13): Protein.Group Protein.Names ... num_NAs baseMean_mpi</span></span>
<span><span class="co">#&gt; colnames(9): neg_ctrl_1 neg_ctrl_2 ... motif2_2 motif2_3</span></span>
<span><span class="co">#&gt; colData names(4): label ID condition replicate</span></span>
<span> <span class="va">mpi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">'median_peptide_intensities'</span><span class="op">)</span></span>
<span> </span>
<span> <span class="va">rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">rd</span><span class="op">$</span><span class="va">baseMean_mpi</span><span class="op">)</span> <span class="co"># shows the average mpi per proteinGroup over all samples</span></span>
<span><span class="co">#&gt; [1] 339967.8 366450.2 411634.6 600430.1 307316.9 108054.2</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="differential-protein-analysis">Differential protein analysis<a class="anchor" aria-label="anchor" href="#differential-protein-analysis"></a>
</h3>
<p>To perform differential protein expression analysis, you have to run
the get_DEPresults() function. There are three main types: ‘manual’ (1
vs 1), ‘control’ (all vs 1), and ‘all’ (all vs all). For ‘manual’, you
can also specify the contrasts you want to test by providing a character
vector for the ‘tests’ parameter.In addition, you can choose the p.adj
cutoff and log2 fold change cutoffs for significant, and the method of
FDR correction. The DEP default is ‘fdrtool’, but this has given some
weird results in the past. Therefore, the default here is ‘BH’
(Benjamini-Hochberg), which is also the default that limma uses (which
DEP uses in the background).</p>
<p>get_DEPresults returns a data frame with statistics for the specified
tests, and can be used for visualization afterwards.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># To test a 1 vs 1 comparison</span></span>
<span><span class="va">res_man</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_DEPresults.html">get_DEPresults</a></span><span class="op">(</span><span class="va">se</span>, condition1 <span class="op">=</span> <span class="st">'motif1'</span>, condition2 <span class="op">=</span> <span class="st">'neg_ctrl'</span>,</span>
<span>                      type <span class="op">=</span> <span class="st">'manual'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Tested contrasts: motif1_vs_neg_ctrl</span></span>
<span></span>
<span><span class="co"># To test multiple 1 vs 1 comparisons</span></span>
<span><span class="va">res_man2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_DEPresults.html">get_DEPresults</a></span><span class="op">(</span><span class="va">se</span>,</span>
<span>                           tests <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'motif1_vs_neg_ctrl'</span>, <span class="st">'motif1_vs_motif2'</span><span class="op">)</span>,</span>
<span>                           type <span class="op">=</span> <span class="st">'manual'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Tested contrasts: motif1_vs_neg_ctrl, motif1_vs_motif2</span></span>
<span></span>
<span><span class="co"># To test all conditions vs 1 reference condition</span></span>
<span><span class="va">res_ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_DEPresults.html">get_DEPresults</a></span><span class="op">(</span><span class="va">se</span>, ref_condition <span class="op">=</span> <span class="st">'neg_ctrl'</span>, type <span class="op">=</span> <span class="st">'control'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Tested contrasts: motif1_vs_neg_ctrl, motif2_vs_neg_ctrl</span></span>
<span></span>
<span><span class="co"># To test all vs all</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_DEPresults.html">get_DEPresults</a></span><span class="op">(</span><span class="va">se</span>, type <span class="op">=</span> <span class="st">'all'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Tested contrasts: neg_ctrl_vs_motif1, neg_ctrl_vs_motif2, motif1_vs_motif2</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="plotting">Plotting<a class="anchor" aria-label="anchor" href="#plotting"></a>
</h2>
<div class="section level3">
<h3 id="volcano-plots">Volcano plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h3>
<p>The plotVolcano() function returns volcano plots with the specified
significance cutoffs. If more than 1 comparison is present in your
results data frame, a list of volcano plots will be returned which can
be accessed by the ‘$’ operator. Check the help page for plotVolcano()
(<code><a href="../reference/plotVolcano.html">?plotVolcano</a></code>) to see the different options for labeling
specific points in the volcano.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plotVolcano.html">plotVolcano</a></span><span class="op">(</span><span class="va">res_man</span><span class="op">)</span> <span class="co"># Default volcano plot if one comparison is present.</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>                    <span class="co"># labels all significant points (can be a bit much).</span></span>
<span></span>
<span></span>
<span><span class="va">volcano_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotVolcano.html">plotVolcano</a></span><span class="op">(</span><span class="va">res_ref</span>, label <span class="op">=</span> <span class="st">''</span><span class="op">)</span> <span class="co"># returns list of volcano plots</span></span>
<span>                                                 <span class="co"># Don't label anything.</span></span>
<span><span class="va">volcano_list</span><span class="op">$</span><span class="va">motif1_vs_neg_ctrl</span> <span class="co"># Select which plot you want to see.</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-17-2.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Example of a very ugly volcano plot.</span></span>
<span><span class="fu"><a href="../reference/plotVolcano.html">plotVolcano</a></span><span class="op">(</span><span class="va">res_man</span>, pval_cutoff <span class="op">=</span> <span class="fl">0.001</span>, fc_cutoff <span class="op">=</span> <span class="fl">2</span>,</span>
<span>            up_color <span class="op">=</span>  <span class="st">'blue'</span>, down_color <span class="op">=</span> <span class="st">'yellow'</span>, ns_color <span class="op">=</span> <span class="st">'green'</span>,</span>
<span>            label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'SMAD3'</span>, <span class="st">'SMAD4'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-17-3.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="ma-plots">MA-plots<a class="anchor" aria-label="anchor" href="#ma-plots"></a>
</h3>
<p>If median_peptide_intensities are added to the se, you can also plot
an MA-plot, with abundances on the x-axis, and fold-changes on the
y-axis. Significant hits are indicated in blue/red:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_MA.html">plot_MA</a></span><span class="op">(</span><span class="va">res_man</span>, label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'SMAD3'</span>, <span class="st">'SMAD4'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text_repel()`).</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="venn-diagrams">Venn diagrams<a class="anchor" aria-label="anchor" href="#venn-diagrams"></a>
</h3>
<p>You can also make a Venn diagram showing overlapping significant
proteins. By default, all comparisons present in your results data frame
are used, but it can handle five comparisons maximally. You can specify
which comparisons to include.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_venn_diagram.html">plot_venn_diagram</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># all comparisons</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [[1]]</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] venn_list</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $filename</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] NULL</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $disable.logging</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] T</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fill</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] colors</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fontfamily</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "sans"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fontface</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "bold"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $cat.fontfamily</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "sans"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $lty</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] 0</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25]</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_venn_diagram.html">plot_venn_diagram</a></span><span class="op">(</span><span class="va">res</span>, comparisons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'neg_ctrl_vs_motif1'</span>, </span>
<span>                                       <span class="st">'neg_ctrl_vs_motif2'</span><span class="op">)</span><span class="op">)</span> <span class="co"># only two comp.</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [[1]]</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] venn_list</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $filename</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] NULL</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $disable.logging</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] T</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fill</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] colors</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fontfamily</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "sans"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fontface</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "bold"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $cat.fontfamily</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "sans"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $lty</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] 0</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25]</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-19-2.png" width="700"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_venn_diagram.html">plot_venn_diagram</a></span><span class="op">(</span><span class="va">res</span>, comparisons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'neg_ctrl_vs_motif1'</span>, </span>
<span>                                       <span class="st">'neg_ctrl_vs_motif2'</span><span class="op">)</span>, </span>
<span>                  colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'red'</span>, <span class="st">'blue'</span><span class="op">)</span><span class="op">)</span> <span class="co"># specify colors used</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [[1]]</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] venn_list</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $filename</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] NULL</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $disable.logging</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] T</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fill</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] colors</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fontfamily</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "sans"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $fontface</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "bold"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $cat.fontfamily</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] "sans"</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] </span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] $lty</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25] [1] 0</span></span>
<span><span class="co">#&gt; INFO [2025-06-18 13:55:25]</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-19-3.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="upset-plot">Upset plot<a class="anchor" aria-label="anchor" href="#upset-plot"></a>
</h3>
<p>As an alternative to Venn diagrams, you can also plot protein
membership as an upset plot. Individual components of the plot can be
altered after the plot is made. Check the ‘ggupset’ vignette for
instructions on how to do this.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">upset_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_upset.html">plot_upset</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="va">upset_plot</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p>Finally, to get an overview of the number of identified significant
proteins per condition:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_DEP_barplot.html">plot_DEP_barplot</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
<p>You can change the conditions included, order of columns, and
labels:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Include only two comparisons and change the order:</span></span>
<span><span class="fu"><a href="../reference/plot_DEP_barplot.html">plot_DEP_barplot</a></span><span class="op">(</span><span class="va">res</span>, comparisons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'neg_ctrl_vs_motif2'</span>,</span>
<span>                                      <span class="st">'neg_ctrl_vs_motif1'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Same as above, but axis labels are changed.</span></span>
<span><span class="fu"><a href="../reference/plot_DEP_barplot.html">plot_DEP_barplot</a></span><span class="op">(</span><span class="va">res</span>, comparisons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'neg_ctrl_vs_motif2'</span>,</span>
<span>                                      <span class="st">'neg_ctrl_vs_motif1'</span><span class="op">)</span>,</span>
<span>                 names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'motif2'</span>, <span class="st">'motif1'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-22-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="gene-set-enrichment-analysis-gsea">Gene set enrichment analysis (GSEA)<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis-gsea"></a>
</h2>
<p>GSEA can be performed on the results from get_DEPresults(). This only
works if the results object contains one comparison. It does not make a
lot of sense to do GSEA on a DNA pull down, but this will be used for
demonstration purposes nonetheless.</p>
<p>In addition to a results object, you need gene sets that you want to
test. For this, you need to download the msigdb superset with gene sets,
and extract the sets that you want. Most commonly, cancer hallmarks (h),
curated gene sets (c2) and gene ontology gene sets (c5) are used. To
extract these:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_msigdb.html">load_msigdb</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="st">'hs'</span><span class="op">)</span> <span class="co"># loads the super set</span></span>
<span><span class="co">#&gt; see ?msigdb and browseVignettes('msigdb') for documentation</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; require("GSEABase")</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Warning in getMsigOrganism(gsc, idType): Assuming the organism to be human.</span></span>
<span></span>
<span><span class="co"># retrieve the cancer hallmarks </span></span>
<span><span class="va">geneset_hallmarks</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genesets.html">get_genesets</a></span><span class="op">(</span><span class="va">db</span>, collection <span class="op">=</span> <span class="st">'h'</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># retrieve the GO:biological process gene sets.</span></span>
<span><span class="co"># similarly molecular function (GOMF) and cellular compartment (GOCC) can be </span></span>
<span><span class="co"># retrieved</span></span>
<span><span class="va">geneset_gobp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genesets.html">get_genesets</a></span><span class="op">(</span><span class="va">db</span>, collection <span class="op">=</span> <span class="st">'c5'</span>, subcollection <span class="op">=</span> <span class="st">'GOBP'</span><span class="op">)</span></span></code></pre></div>
<p>To perform GSEA, use the results object and the gene sets of
choice:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_DEPresults.html">get_DEPresults</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">'motif1'</span>, <span class="st">'neg_ctrl'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Tested contrasts: motif1_vs_neg_ctrl</span></span>
<span><span class="va">gsea</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/perform_GSEA.html">perform_GSEA</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">geneset_gobp</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.04% of the list).</span></span>
<span><span class="co">#&gt; The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre></div>
<div class="section level3">
<h3 id="plotting-of-gsea-results">Plotting of GSEA results<a class="anchor" aria-label="anchor" href="#plotting-of-gsea-results"></a>
</h3>
<p>There are several options to visualize your GSEA results. The
simplest is with a bar plot, showing just the Normalized Enrichment
Scores (NES). The default options shows all significant results. Be
specifying the top_n parameter, you can only include the n pathways with
the lowest padj value.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Bar plot of 10 pathways with lowest padj values.</span></span>
<span><span class="va">barplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_gsea_barplot.html">plot_gsea_barplot</a></span><span class="op">(</span><span class="va">gsea</span>, top_n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">barplot</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-25-1.png" width="700"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#Bar plot of 10 pathways with lowest values and different colors for bars.</span></span>
<span><span class="va">barplot2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_gsea_barplot.html">plot_gsea_barplot</a></span><span class="op">(</span><span class="va">gsea</span>, pos_color <span class="op">=</span> <span class="st">'red3'</span>, </span>
<span>                              neg_color <span class="op">=</span> <span class="st">'dodgerblue'</span>, top_n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">barplot2</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-25-2.png" width="700"></p>
<p>With the plot_gsea_dotplot() function, you can include more
information about the pathways:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">dotplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_gsea_dotplot.html">plot_gsea_dotplot</a></span><span class="op">(</span><span class="va">gsea</span>, top_n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dotplot</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-26-1.png" width="700"></p>
<p>A third way way to visualize the results from GSEA is with a volcano
plot with the NES on the x-axis and -log10(padj) on the y-axis. Like
with the volcano plot for DEP visualization, you can specify how to
label points.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot all significant points (can be chatoic)</span></span>
<span><span class="va">volcano</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_gsea_volcano.html">plot_gsea_volcano</a></span><span class="op">(</span><span class="va">gsea</span><span class="op">)</span></span>
<span></span>
<span><span class="va">volcano</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-27-1.png" width="700"></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># plot 10 significant points with the lowest padj value on both sides of volcano.</span></span>
<span><span class="va">volcano</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_gsea_volcano.html">plot_gsea_volcano</a></span><span class="op">(</span><span class="va">gsea</span>, top_n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">volcano</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-27-2.png" width="700"></p>
<p>If you ran multiple GSEAs on different results objects (from
get_DEPresults()), you can plot them in a single bubble plot.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">res1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_DEPresults.html">get_DEPresults</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">'motif1'</span>, <span class="st">'neg_ctrl'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Tested contrasts: motif1_vs_neg_ctrl</span></span>
<span><span class="va">res2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_DEPresults.html">get_DEPresults</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">'motif2'</span>, <span class="st">'neg_ctrl'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Tested contrasts: motif2_vs_neg_ctrl</span></span>
<span></span>
<span><span class="va">gsea1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/perform_GSEA.html">perform_GSEA</a></span><span class="op">(</span><span class="va">res1</span>, <span class="va">geneset_gobp</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.04% of the list).</span></span>
<span><span class="co">#&gt; The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span>
<span><span class="va">gsea2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/perform_GSEA.html">perform_GSEA</a></span><span class="op">(</span><span class="va">res2</span>, <span class="va">geneset_gobp</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.02% of the list).</span></span>
<span><span class="co">#&gt; The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span>
<span></span>
<span><span class="va">bubble</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_gsea_bubbleplot.html">plot_gsea_bubbleplot</a></span><span class="op">(</span><span class="va">gsea1</span>, <span class="va">gsea2</span>,</span>
<span>                               sample_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'motif1'</span>, <span class="st">'motif2'</span><span class="op">)</span>, top_n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bubble</span></span>
<span><span class="co">#&gt; Warning: Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span></code></pre></div>
<p><img src="DIANNmv_files/figure-html/unnamed-chunk-28-1.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
